test/index.test.ts

```diff
@@ -1,51 +1,103 @@
-  it("should be hexadecimal", () => {
-    expect(hash.substring(0, 2)).toBe("0x");
-  });
-  it("should create a hash of 32 bites", () => {
-    expect(hash.length).toBe(66);
+  dataTest.forEach((data) => {
+    const hash = lsp6Signer.hashMessage(data);
+    it("should be hexadecimal", () => {
+      expect(hash.substring(0, 2)).toBe("0x");
+    });
+    it("should create a hash of 32 bites", () => {
+      expect(hash.length).toBe(66);
+    });
   });
 });
 
 describe("Sign transaction function", () => {
   const lsp6Signer = new LSP6Signer();
-  const signedObject = lsp6Signer.sign("hello", signingKey);
-  it("should give back an object with the right properties", () => {
-    expect(signedObject).toHaveProperty("message");
-    expect(signedObject).toHaveProperty("messageHash");
-    expect(signedObject).toHaveProperty("v");
-    expect(signedObject).toHaveProperty("r");
-    expect(signedObject).toHaveProperty("s");
-    expect(signedObject).toHaveProperty("signature");
+  dataTest.forEach((data) => {
+    const signedObject = lsp6Signer.sign(data, signingKey);
+    it("should give back an object with the right properties", () => {
+      expect(signedObject).toHaveProperty("message");
+      expect(signedObject).toHaveProperty("messageHash");
+      expect(signedObject).toHaveProperty("v");
+      expect(signedObject).toHaveProperty("r");
+      expect(signedObject).toHaveProperty("s");
+      expect(signedObject).toHaveProperty("signature");
+    });
   });
 });
 
 describe("Recover the address function of a transaction", () => {
-  const lsp6Signer = new LSP6Signer();
-  const signingAddress = "0x2b389f8EB52D16A105e02165a2AC1450461A237b"; //private key
-  const messageInfo = lsp6Signer.sign("hello", signingKey);
-  const message = messageInfo.message;
-  const signature = messageInfo.signature;
-  const isMessagePrefixed = false;
-  const recoveredAddress = lsp6Signer.recover(
-    message,
-    signature,
-    isMessagePrefixed
-  );
-  it("should recover the signing address of a transaction", () => {
-    expect(recoveredAddress).toBe(signingAddress);
+  it("should recover the signing address of a transaction when the message is not prefixed", () => {
+    const lsp6Signer = new LSP6Signer();
+    dataTest.forEach((data) => {
+      const messageData = lsp6Signer.sign(data, signingKey);
+      const message = messageData.message;
+      const signature = messageData.signature;
+      const isMessagePrefixed = false;
+      const recoveredAddress = lsp6Signer.recover(
+        message,
+        signature,
+        isMessagePrefixed
+      );
+      expect(recoveredAddress).toBe(signingAddress);
+      expect(recoveredAddress.substring(0, 2)).toBe("0x");
+      expect(recoveredAddress.length).toBe(42);
+    });
   });
-  it("should be hexadecimal", () => {
-    expect(recoveredAddress.substring(0, 2)).toBe("0x");
+
+  it("should recover the signing address of a transaction when the message is prefixed", () => {
```

Comment:
This is not really when the message is prefixed, as the has function is doing the prefixing for us here. It is more like 'should recover the signer address when recovering signed hash'